census/census.go

```diff
@@ -99,20 +141,63 @@ func (m *Manager) Init(storageDir, rootAuthPubKey string) error {
 			cns.RootKey = rootAuthPubKey
 		}
 		m.Census = cns
-		if err := os.WriteFile(filepath.Clean(nsConfig), []byte(""), 0o600); err != nil {
+		if err := os.WriteFile(nsConfig, []byte(""), 0o600); err != nil {
 			return err
 		}
 		return m.save()
 	}
 
-	jsonBytes, err := os.ReadFile(filepath.Clean(nsConfig))
+	jsonBytes, err := os.ReadFile(nsConfig)
 	if err != nil {
 		return err
 	}
-	if err := json.Unmarshal(jsonBytes, &m.Census); err != nil {
+	var namespaces Namespaces
+	if err := json.Unmarshal(jsonBytes, &namespaces); err != nil {
 		log.Warn("could not unmarshal json config file, probably empty. Skipping")
 		return nil
 	}
+
+	// Copy censuses of the expected version into m.Census and remove old ones
+	m.Census.RootKey = namespaces.RootKey
+	cleanOldCensues := false
+	now := time.Now()
+	for _, v := range namespaces.Namespaces {
+		switch v.Version {
+		case 0:
+			// Version 0 are graviton censuses.  These censuses are
+			// no longer supported and belong to legacy processes.
+			cleanOldCensues = true
+			outdatedDir := filepath.Join(storageDir,
+				fmt.Sprintf("outdated.%v", now.Format(time.RFC3339)), "v0")
+			if err := os.MkdirAll(outdatedDir, 0o700); err != nil {
+				return err
+			}
+			log.Infof("Moving outdated census %v to %v", v.Name, outdatedDir)
+			dir := strings.Split(v.Name, string(os.PathSeparator))[0]
```

Comment:
a bit easier to use filepath.SplitList. Or, if there's just one separator, filepath.Dir is the equivalent of getting the parent directory.

Reply:
Ah! That's what I thought!  But then realized that `SplitList` separates by `:` in Unix, and it's meant to separate lists of paths found in `$PATH` for example.
In this case there are either 1 or 2 elements in the Name, and `Dir` would only work when there are 2 elements.  I just wanted a function that returns the list of elements in a path, and didn't find such function, so I used `strings.Split` with `os.PathSeparator`

Reply:
Oops, you're absolutely right.

How about:

```
if dir := filepath.Dir(v.Name); dir != "." {
    v.Name = dir
}
```

Slightly longer, but also more obvious with its logic, if you ask me.

Reply:
I botched my example above, yeah, it shouldn't be updating `v.Name`. Your method sounds fine.